asm/amd/insn.go

```diff
@@ -0,0 +1,18 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package amd // import "go.opentelemetry.io/ebpf-profiler/asm/amd"
+
+// IsEndbr64 returns true if the first 4 bytes of the code is endbr64 instruction
+// https://www.felixcloutier.com/x86/endbr64
+// The second returned argument is the size of the instruction which is always 4
+func IsEndbr64(code []byte) (isEndbr bool, size int) {
```

Comment:
Could this instead just a `Decode`, `DecodeSkippable` or similar? I suspect we may need to add few more here...